app/controllers/api/v1/workflows_controller.rb

```diff
@@ -81,7 +83,9 @@ def add_relation(resource, relation, value)
   def new_items(resource, relation, value)
     case relation
     when :retired_subjects, 'retired_subjects'
+      resource.save!
       value.flat_map {|id| resource.retire_subject(id) }
+      resource.reload
```

Comment:
also we can modify the yield block in `updatable_resource#update_links` to allow a forced reload of the workflow but we currently use it for post persistence operations so we'd have to deal with that.